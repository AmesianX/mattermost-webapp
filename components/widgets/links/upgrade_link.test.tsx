// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {shallow} from 'enzyme';

import UpgradeLink from './upgrade_link';

describe('components/widgets/links/UpgradeLink', () => {
    const handleClick = jest.fn();

    test('should match the snapshot on show', () => {
        const wrapper = shallow(
            <UpgradeLink handleClick={handleClick}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should trigger telemetry call when button clicked', (done) => {
        const wrapper = shallow(
            <UpgradeLink handleClick={handleClick}/>,
        );
        expect(wrapper.find('button').exists()).toEqual(true);
        wrapper.find('button').simulate('click');

        setImmediate(() => {
            expect(handleClick).toHaveBeenCalled();
            done();
        });
        expect(wrapper).toMatchSnapshot();
    });
});
