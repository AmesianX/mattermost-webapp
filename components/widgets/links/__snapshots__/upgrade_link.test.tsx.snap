// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/links/UpgradeLink should match the snapshot on show 1`] = `
<button
  className="upgradeLink"
  onClick={[Function]}
>
  <FormattedMessage
    defaultMessage="Upgrade now"
    id="upgradeLink.warn.upgrade_now"
  />
</button>
`;

exports[`components/widgets/links/UpgradeLink should trigger telemetry call when button clicked 1`] = `
<button
  className="upgradeLink"
  onClick={[Function]}
>
  <FormattedMessage
    defaultMessage="Upgrade now"
    id="upgradeLink.warn.upgrade_now"
  />
</button>
`;
